0000-product.md

```diff
@@ -0,0 +1,463 @@
+- Feature Name: Product
+- Start Date: 5/13/19
+- RFC PR: [#5](https://github.com/hyperledger/grid-rfcs/pull/5)
+
+# Summary
+[summary]: #summary
+
+
+This RFC proposes a generic and extensible framework for a Hyperledger Grid
+Product entity as well as a more specific - and fully encapsulated - GS1
+compliant Product.
+
+GS1 is a widely used data standard in enterprises and, given that positioning
+and familiarity, Grid support feels natural.  Additional implementations of
+Product for specialized industries or use cases may derive from or extend this
+implementation.
+
+_For the base implementation of a product on grid, there will be 4 fields. A
+product_id, a product_namespace, an owner, and properties (a repeated key-value
+field)._
+
+
+# Motivation
+[motivation]: #motivation
+
+The Grid Product implementation is designed for sharing product master data
+between participants. Product is a near universal concept within supply chain
+solutions and would naturally be one of the highest areas of re-use across Grid
+applications.
+
+The desgn will address use cases including:
+
+- Sharing of Product master data across a network
+- Including Product in other business transactions (track and trace events,
+  purchases, sales, etc)
+- Enriching UX experiences by including additional attribution of a Product such
+  as names or descriptions
+
+It is also useful to use product information as auxiliary data in other supply
+chain solutions.  For example, in track and trace, the location and temperature
+of an item (an instance of a product) is stored.  When presenting this
+information later to a human user, it is useful to combine it with information
+about the product, such as the product’s name and description.  Therefore,
+Product becomes a reusable and important component within Grid.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A **__product__** is an archetype of an item that is transacted, traded, or
+referenced in a supply chain.  Each product has a **__product_namespace__**.
+This RFC defines a single product_namespace for GS1; a product in the GS1
+product_namespace is called a GS1 product.  Note that this design supports
+extending additional product namespaces in the future.
+
+A product is referenced using a **product_id**.  For GS1 products, the
+product_id is a Global Trade Item Number (GTIN), which is part of the GS1
+specifications.
```

Comment:
Are we saying "product' is limited to item level and not higher levels of packaging (like cases or pallets that may use another GS1 key??) or even assets (GIAI/GRAI?) OR any lower granularity of a GTIN such as batch/lot or serial?

Reply:
I wouldn't say it's limited, I think of this as information (at an item level) that can be leveraged to build-out those "higher level" constructs like cases or pallets. Same idea with assets. The scope of those things would better be captured in another RFC. 

What do you mean by the lower granularity of a GTIN?

Reply:
Serialized GTIN or GTIN + batch lot

Reply:
So instead of just describing a 12 oz bottle of water, you are describing a more specific bottle of water, which may have a very different origin point,. expiry date, etc.

Reply:
Gotcha. So using grid product you could define a schema that includes SGTIN, such that an application could perform actions on it at an individual "unit" level

Reply:
> Gotcha. So using grid product you could define a schema that includes SGTIN, such that an application could perform actions on it at an individual "unit" level

I'm not suggested this is the best practice or pattern, just stating that it would be possible to implement this functionality in such a manner with grid product as it is defined in this RFC.

Reply:
I think we would created a different class of records for instances that are 'of a type' of a product instance. That way, the bulk of the product data would not be replicated on the instance records.

Reply:
`class` is an ambiguous term here - I assume you are talking about a different schema and or a different product namespace?

Reply:
Yes, I'm saying that instance data should not be considered part of the product record (there is reference to things like temp data and expiration date below). In my opinion, these should be attributes on instance-level records that share the GTIN portion with the Product records that contains all the common (cross-instance) attributes.

Reply:
Fully aligned w/ that.